index.html

```diff
@@ -1,390 +1,1451 @@
+                <dfn>`BufferOverrunError`</dfn> on the [=serial port task
+                source=] and invoke the steps to [=handle closing the readable
+                stream=].
+              <li>
+                If a break condition was encountered, [=queue a task=] to  invoke
+                |controller|.{{ReadableStreamDefaultController/error()}} with a
+                <dfn>`BreakError`</dfn> on the [=serial port task source=] and
+                invoke the steps to [=handle closing the readable stream=].
+              <li>
+                If a framing error was encountered, [=queue a task=] to invoke
+                |controller|.{{ReadableStreamDefaultController/error()}} with a
+                <dfn>`FramingError`</dfn> on the [=serial port task source=] and
+                invoke the steps to [=handle closing the readable stream=].
+              <li>
+                If a parity error was encountered, [=queue a task=] to invoke
+                |controller|.{{ReadableStreamDefaultController/error()}} with a
+                <dfn>`ParityError`</dfn> on the [=serial port task source=] and
+                invoke the steps to [=handle closing the readable stream=].
+              <li>
+                If an operating system error was encountered, [=queue a task=] to invoke
+                |controller|.{{ReadableStreamDefaultController/error()}} with a
+                <dfn>`UnknownError`</dfn> on the [=serial port task source=] and
+                invoke the steps to [=handle closing the readable stream=].
+              <li>
+                If the port was disconnected, [=queue a task=] to invoke
+                |controller|.{{ReadableStreamDefaultController/error()}} with a
+                {{NetworkError}} on the [=serial port task source=], set
+                [=this=].{{[[readFatal]]}} to `true`, and invoke the steps to
+                [=handle closing the readable stream=].
+            </ol>
+        </ol>
+
+      <p class="note">
+        The {{Promise}} returned by this algorithm is immediately resolved so
+        that it does not block canceling the stream. [[STREAMS]] specifies that
+        this algorithm will not be invoked again until a chunk is enqueued.
+      </p>
+
+      <li>
+        Let |cancelAlgorithm| be the following steps:
+        <ol>
+          <li>Let |promise| be [=a new promise=].
+          <li>
+            Run the following steps [=in parallel=].
+            <ol>
+              <li>Invoke the operating system to discard the contents of all
+              software and hardware receive buffers for the port.
+              <li>Invoke the steps to [=handle closing the readable stream=].
+              <li>
+                [=Queue a task=] to resolve |promise| on the [=serial port task
+                source=].
+            </ol>
+          <li>Return |promise|.
+        </ol>
+
+      <li>
+        [=ReadableStream/Set up=] |stream| with
```

Comment:
I was going to suggest you use "create" instead of new + "set up", but I seem to have messed that up on the streams side: https://github.com/whatwg/streams/issues/1084